crypto/x509/x_pubkey.c

```diff
@@ -69,27 +70,51 @@ int X509_PUBKEY_set(X509_PUBKEY **x, EVP_PKEY *pkey)
     if ((pk = X509_PUBKEY_new()) == NULL)
         goto error;
 
-    if (pkey != NULL && pkey->ameth) {
-        if (pkey->ameth->pub_encode) {
-            if (!pkey->ameth->pub_encode(pk, pkey)) {
-                X509err(X509_F_X509_PUBKEY_SET,
-                        X509_R_PUBLIC_KEY_ENCODE_ERROR);
+    if (pkey != NULL) {
```

Comment:
This might be better as

````
if (pkey == NULL)
    goto unsupported;
````
Then all following lines can be moved to the left by one indent.

Reply:
Okie

Reply:
That was a good exercise, it had me discover a probably memory leak...